lib/edgehog_device_forwarder/http_requests/core.ex

```diff
@@ -0,0 +1,45 @@
+defmodule EdgehogDeviceForwarder.HTTPRequests.Core do
+  alias EdgehogDeviceForwarderProto.Edgehog.Device.Forwarder.Http, as: HTTP
+
+  @table EdgehogDeviceForwarder.Supervisors.Caches.for(EdgehogDeviceForwarder.HTTPRequests)
+
+  @type controller :: pid()
+  @type id :: binary()
+
+  @spec forward(controller, id, HTTP.Response.t()) :: :ok
+  def forward(controller, request_id, response) do
+    action = response_action(response)
+
+    case action do
+      :respond ->
+        ConCache.dirty_delete(@table, request_id)
+
+      {:upgrade, :websocket} ->
+        ConCache.dirty_put(@table, request_id, {:upgrading, :websocket, Qex.new()})
+    end
```

Comment:
Are these `dirty` because you're inside an `isolated` block? In that case you're leaking this implicit assumption inside this separate module, so you should probably move this stuff in the parent module and possibly leave only the pure response generation here.

Reply:
All ConCache functions in websockets/core and http_requests/core are dirty.

from the websockets.core moduledoc:

>   All of its methods are meant to be used only by the WebSockets module and are meant to be run
  inside an isolated ConCache environment.

right now core/error modules describe what to do in the happy/sad paths, while the parent module performs the checks needed.